### ISSUE TITLE
[0.4.6] restrict-plus-operands misses any values and return types imported from another file

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1285]
## Summary

With explicit allowAny:false, @typescript-eslint/restrict-plus-operands detects an any-returning function declared in the same file, but misses the identical function after importing it from another TypeScript file. An imported constant explicitly annotated any also escapes the check.

The local-function and local type assertion controls are reported correctly; number controls stay clean. The imported-function and imported-constant misses persist when both example.tsx and reader.ts are passed to the native CLI together.

This is a cross-file boundary, separate from the original same-file examples in #1273 and #1277, which now pass in 0.4.6.

Verified with @utoo/lint 0.4.6, ESLint 8.57.1, @typescript-eslint/parser and eslint-plugin 6.21.0, Node.js 22.23.1 on macOS arm64. There are no parser failures or suppression comments, and only this rule is enabled.

## Setup

```sh
npm init -y
npm install --save-dev @utoo/lint@0.4.6 eslint@8.57.1 @typescript-eslint/parser@6.21.0 @typescript-eslint/eslint-plugin@6.21.0 typescript@5.3.3
```

`tsconfig.json`:

```json
{"compilerOptions":{"strict":true,"target":"ES2022","jsx":"react"},"include":["*.tsx","*.ts"]}
```

`.eslintrc.cjs`:

```js
module.exports = {
  "root": true,
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2022,
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
};
```

`utlint.config.json`:

```json
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
        "allowAny": false
      }
    ]
  }
}
```

Run each case separately, replacing the files with the corresponding snippet:

```sh
./node_modules/.bin/eslint --no-eslintrc --config .eslintrc.cjs --format=json example.tsx
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx
# For cases with reader.ts, including both files also reproduces:
./node_modules/.bin/utlint --config utlint.config.json --format=json example.tsx reader.ts
```

## imported-return-any

`example.tsx`:

```tsx
import {read} from "./reader";export const result=read()+1;
```

`reader.ts`:

```ts
export function read():any{return 1;}
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## local-return-any

`example.tsx`:

```tsx
function read():any{return 1;}export const result=read()+1;
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## imported-return-number

`example.tsx`:

```tsx
import {read} from "./reader";export const result=read()+1;
```

`reader.ts`:

```ts
export function read():number{return 1;}
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

## local-return-number

`example.tsx`:

```tsx
function read():number{return 1;}export const result=read()+1;
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

## imported-any-annotated-local

`example.tsx`:

```tsx
import {read} from "./reader";export const result=(read() as any)+1;
```

`reader.ts`:

```ts
export function read():any{return 1;}
```

ESLint: 1 diagnostic(s). utlint: 1 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## imported-const-any

`example.tsx`:

```tsx
import {value} from "./reader";export const result=value+1;
```

`reader.ts`:

```ts
export const value:any=1;
```

ESLint: 1 diagnostic(s). utlint: 0 diagnostic(s).

ESLint message: Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `boolean`, `null`, `RegExp`, `undefined`. Got `any`.

## imported-const-number

`example.tsx`:

```tsx
import {value} from "./reader";export const result=value+1;
```

`reader.ts`:

```ts
export const value:number=1;
```

ESLint: 0 diagnostic(s). utlint: 0 diagnostic(s).

Expected: preserve the explicitly declared any types across imports and report their addition when allowAny:false, while retaining the clean numeric controls.
